components/pedigree/resources/src/main/resources/pedigree/filesaver/FileSaver.js

```diff
@@ -238,12 +238,13 @@ define([
   // with an attribute `content` that corresponds to the window
 
   var saveTextAs = saveTextAs
-  || (function (textContent, fileName, charset) {
+  || (function (textContent, fileName, charset, type) {
       fileName = fileName || 'pedigree.txt';
       charset = charset || 'utf-8';
       textContent = (textContent || '').replace(/\r?\n/g, "\r\n");
+      type = (type || "text/plain;charset=") + charset;
```

Comment:
This assumes that the `type` argument ends with `;charset=`, which isn't always the case. Either put the brackets around the last 2 things instead of the first 2, or add a check if the value so far ends with `charset=`.